internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +95,54 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+	if err != nil {
+		return nil, err
+	}
+
+	disks := int64(getValidDiskCount(inventory.Disks) - 1)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	}
+
+	if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
```

Comment:
@gamli75 do we need to do the role check for 3 hosts? Can there be a case when 2 hosts will have role `master` and 1 will have role `worker` ?

Reply:
I think we can keep it even if it's redundant - don't think it's critical